oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/converter/ObjectToListStringConverter.java

```diff
@@ -59,7 +59,7 @@ public Object convert(Object source, TypeDescriptor sourceType, TypeDescriptor t
 		if (source instanceof List) {
```

Comment:
In my opinion this check can be removed, as it will be covered by ```if (source instanceof Collection) ``` where a new ArrayList is created

Reply:
Also the code current code does not cover the case of  a `List<Object>` where the first element is a String but the other elements are not. If this code block is removed it will be handled correctly by the `if (source instanceof Collection)` block which converts all elements to a `String`.

Reply:
I agree. This is what I did in my initial implementation, but I wanted to double check if someone else is sharing the same opinion. I will apply the stash which contains this.